include/vctr/Containers/VctrBase.h

```diff
@@ -247,6 +247,28 @@ class VctrBase : private Config,
         std::copy_n (otherData, otherSize, begin());
     }
 
+    /** Fills the container with the given value. */
+    constexpr void fill (const value_type& value) {  std::fill (storage.begin(), storage.end(), value); }
```

Comment:
Moving fill to base is absolutely correct since this should be available to all container types. Please note however that `storage.begin()` and `storage.end()` might fill more than needed as we are using overly large arrays under the hood to have memory regions that are always a multiple of a simd register size. In this particular case, the worst thing that can happen is that we write to unused dummy memory, which is safe, the best thing that can happen is that the compiler makes use of this over-allocated memory and uses some simd acceleration to speed this up. Not sure about the cleverness of the compiler though ;) 

TLDR: Usually, you should always use `begin()` and `end()` in VctrBase instead of `storage.begin()` and `storage.end()` except you really know what you are doing. In this particular case this is okay

Reply:
Thanks for this very useful comment.

Reply:
Using begin/end now.